### PR TITLE
Add ImageContentSourcePolicy CRD manifest

### PR DESCRIPTION
**Pull request by umohnani8**
Adding a ImageContentSourcePolicy manifest to create cluster wide
registry mirror configurations.

API PR: https://github.com/openshift/api/pull/354

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@abhinavdahiya @mrunalp @mtrmac PTAL

**Comment by umohnani8** [comment]
https://github.com/openshift/api/pull/354 and https://github.com/openshift/client-go/pull/109 are in. This is ready to merge once tests are green!
cc @abhinavdahiya @smarterclayton @mtrmac @mrunalp

**Comment by abhinavdahiya** [comment]
from [e2e-aws](https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-config-operator/67/pull-ci-openshift-cluster-config-operator-master-e2e-aws/321) job [install-failed-during-bootstrap](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-config-operator/67/pull-ci-openshift-cluster-config-operator-master-e2e-aws/321/artifacts/e2e-aws/installer/log-bundle-20190627182247.tar) logs
error while pushing to cluster from bootkube.sh
```
Jun 27 17:52:23 ip-10-0-1-184 bootkube.sh[1395]: Failed to create "0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml" customresourcedefinitions.v1beta1.apiextensions.k8s.io/image-content-source-policy.config.openshift.io -n : CustomResourceDefinition.apiextensions.k8s.io "image-content-source-policy.config.openshift.io" is invalid: [metadata.name: Invalid value: "image-content-source-policy.config.openshift.io": must be spec.names.plural+"."+spec.group, spec.names.plural: Invalid value: "imageContentSourcePolicies": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?'), spec.names.singular: Invalid value: "imageContentSourcePolicy": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')]
```

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: image-content-source-policy.config.openshift.io
```
Comment (abhinavdahiya) [inline manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: the name should be `imagecontentsourcepolicies.operator.openshift.io`

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+spec:
+  group: config.openshift.io
+  scope: Cluster
+  names:
+    kind: ImageContentSourcePolicy
+    singular: imageContentSourcePolicy
```
Comment (abhinavdahiya) [inline manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: all small-case

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: image-content-source-policy.config.openshift.io
+spec:
+  group: config.openshift.io
```
Comment (abhinavdahiya) [inline manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: `operator.openshift.io`

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  scope: Cluster
+  names:
+    kind: ImageContentSourcePolicy
+    singular: imageContentSourcePolicy
+    plural: imageContentSourcePolicies
```
Comment (abhinavdahiya) [inline manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: all small-case

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    listKind: ImageContentSourcePolicyList
+  versions:
+  - name: v1
```
Comment (abhinavdahiya) [inline manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: v1alpha1

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml (hunk 1 above)
Comment (umohnani8) [reply manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: fixed

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml (hunk 2 above)
Comment (umohnani8) [reply manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: fixed

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml (hunk 3 above)
Comment (umohnani8) [reply manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: fixed

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml (hunk 4 above)
Comment (umohnani8) [reply manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: fixed

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml (hunk 5 above)
Comment (umohnani8) [reply manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: fixed

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml (hunk 1 above)
Comment (abhinavdahiya) [reply manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: its still `imagecontentsourcepolicies.config.openshift.io`, it should be `imagecontentsourcepolicies.operator.openshift.io`

error from bootkube.sh 
```console
Jun 27 20:09:24 ip-10-0-11-139 bootkube.sh[1413]: "0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml": failed to create: CustomResourceDefinition.apiextensions.k8s.io "imagecontentsourcepolicies.config.openshift.io" is invalid: metadata.name: Invalid value: "imagecontentsourcepolicies.config.openshift.io": must be spec.names.plural+"."+spec.group
```

**Comment by abhinavdahiya** [comment]
1. https://github.com/openshift/cluster-config-operator/pull/67#discussion_r298371371

2. this is probably needed...

```
glide up -v ## to get the latest struct.
make update-codegen ## update the crds.
```

File: manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml (hunk 1 above)
Comment (umohnani8) [reply manifests/0000_10_config-operator_01_imagecontentsourcepolicy.crd.yaml]: oh, oops. Fixed now.

**Comment by umohnani8** [comment]
@abhinavdahiya ran glide up -v and make update-codegen

**Pull request merged**